### PR TITLE
feat(V2): RHINENG-14034 filter test_results by tags

### PR DESCRIPTION
**Pull request by LightOfHeaven1994**
tested locally with curl and works as expected:
```
RHINENG-14034 ~/Documents/WORK/DEVS/compliance-backend> curl -k -H "Content-Type: application/json" -H "X-RH-IDENTITY: $RHI_IDENTITY" -X GET http://localhost:3000/api/compliance/v2/reports/c333c1ac-da9f-4648-8949-9d90f2548e6b/test_results\?tags\=parsing/protocol\=open-source | jq 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1096  100  1096    0     0   8034      0 --:--:-- --:--:-- --:--:--  8058
{
  "data": [
    {
      "id": "d8928a56-5e45-4f51-8b97-9e010679af6e",
      "end_time": "2024-11-08T10:42:29.973Z",
      "failed_rule_count": 0,
      "supported": true,
      "score": 65.11627906976744,
      "type": "test_result",
      "display_name": "thompson.example",
      "groups": [],
      "tags": [
        {
          "key": "transmitter",
          "value": "neural",
          "namespace": "copying"
        },
        {
          "key": "application",
          "value": "open-source",
          "namespace": "indexing"
        },
        {
          "key": "pixel",
          "value": "solid state",
          "namespace": "generating"
        },
        {
          "key": "driver",
          "value": "multi-byte",
          "namespace": "compressing"
        },
        {
          "key": "protocol",
          "value": "open-source",
          "namespace": "parsing"
        }
      ],
      "os_major_version": 7,
      "os_minor_version": 8,
      "compliant": false,
      "system_id": "d704a587-5386-420e-8f67-0f7da271adb5",
      "security_guide_version": "0.1.46"
    }
  ],
  "meta": {
    "total": 1,
    "tags": [
      "parsing/protocol=open-source"
    ],
    "limit": 10,
    "offset": 0
  },
  "links": {
    "first": "/api/compliance/v2/reports/c333c1ac-da9f-4648-8949-9d90f2548e6b/test_results?limit=10&offset=0&tags=parsing%2Fprotocol%3Dopen-source",
    "last": "/api/compliance/v2/reports/c333c1ac-da9f-4648-8949-9d90f2548e6b/test_results?limit=10&offset=0&tags=parsing%2Fprotocol%3Dopen-source"
  }
}

```

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by skateman** [comment]
If we're setting up tagging on another model, it does not make sense to have a complex logic in resolving taggability. Could you please set up the taggable method in the System model as well and remove the alternative logic [here](https://github.com/RedHatInsights/compliance-backend/blob/master/app/controllers/concerns/tag_filtering.rb#L39)?

**Comment by LightOfHeaven1994** [comment]
pr check has some issues not related to PR
```
14:14:53 [2024-11-11T13:14:53.833Z] 2024-11-11 13:14:49 [    INFO] [          MainThread] some resources failed to become ready: cyndipipeline/compliance
```

File: app/models/v2/test_result.rb [hunk 1]
```diff
@@ -7,7 +7,13 @@ class TestResult < ApplicationRecord
     self.table_name = :v2_test_results
     self.primary_key = :id
 
-    belongs_to :system, class_name: 'V2::System', optional: true
+    if Rails.env.test? # For testing taggable
```
Comment (romanblanco) [inline app/models/v2/test_result.rb:10]: @LightOfHeaven1994 This should probably be removed

**Review by romanblanco — APPROVED** [review]
LGTM :+1:

File: app/models/v2/test_result.rb (hunk 1 above)
Comment (LightOfHeaven1994) [reply app/models/v2/test_result.rb:10]: @romanblanco we did it on purpose as we don't want to set autosave and delegate thing for prod, this is needed for testing only to make a trick

Comment (romanblanco) [reply app/models/v2/test_result.rb:10]: Makes sense. Thanks!

**Pull request closed without merging**